src/path.ts

```diff
@@ -0,0 +1,187 @@
+import { Maybe } from './common/types'
+
+
+export enum Branch {
+  Public = 'public',
+  Pretty = 'p',
+  Private = 'private',
+  PrivateLog = 'privateLog',
+  Version = 'version'
+}
+
+export type Path = string[]
+
+export type DirectoryPath = { directory: Path }
+export type FilePath = { file: Path }
+
+/**
+ * The primarily used type for paths.
+ */
+export type DistinctivePath = DirectoryPath | FilePath
+
+
+
+// CREATION
+
+
+/**
+ * Utility function to create a `DirectoryPath`
+ */
+export function directory(...args: Path): DirectoryPath {
+  if (args.some(p => p.includes("/"))) {
+    throw new Error("Forward slashes `/` are not allowed")
+  }
+  return { directory: args }
+}
+
+/**
+ * Utility function to create a `FilePath`
+ */
+export function file(...args: Path): FilePath {
+  if (args.some(p => p.includes("/"))) {
+    throw new Error("Forward slashes `/` are not allowed")
+  }
+  return { file: args }
+}
+
+/**
+ * Utility function to create a root `DirectoryPath`
+ */
+export function root(): DirectoryPath {
+  return { directory: [] }
+}
+
+
+
+// POSIX
+
+
+/**
+ * Transform a string into a `DistinctivePath`.
+ *
+ * Directories should have the format `path/to/dir/` and
+ * files should have the format `path/to/file`.
+ *
+ * Leading forward slashes are removed too, so you can pass absolute paths.
+ */
+export function fromPosix(path: string): DistinctivePath {
+  const split = path.replace(/^\/+/, "").split("/")
+  if (path.endsWith("/")) return { directory: split.slice(0, -1) }
+  return { file: split }
+}
+
+/**
+ * Transform a `DistinctivePath` into a string.
+ *
+ * Directories will have the format `path/to/dir/` and
+ * files will have the format `path/to/file`.
+ */
+export function toPosix(
+  path: DistinctivePath,
+  { absolute }: { absolute: boolean } = { absolute: false }
+): string {
+  const prefix = absolute ? "/" : ""
+  const joinedPath = unwrap(path).join("/")
+  if (isDirectory(path)) return prefix + joinedPath + (joinedPath.length ? "/" : "")
+  return prefix + joinedPath
+}
```

Comment:
Not sure that makes sense.
I'd like to encourage people to make paths via `wn.path.directory()` and `wn.path.file()`.

And the POSIX format should be set in stone, so for example, if you would say:
```js
fileFromPosix("path/to/file/")
```
That doesn't make sense, because the POSIX format doesn't end a file path with a `/`